test/hotspot/jtreg/gc/stringdedup/TestStringDeduplicationTools.java

```diff
@@ -89,21 +93,58 @@ private static void doFullGc(int numberOfTimes) {
         }
     }
 
+    private static volatile int gcCount;
+    private static NotificationListener listener = new NotificationListener() {
+        @Override
+        public void handleNotification(Notification n, Object o) {
+            if (n.getType().equals(GarbageCollectionNotificationInfo.GARBAGE_COLLECTION_NOTIFICATION)) {
+                GarbageCollectionNotificationInfo info = GarbageCollectionNotificationInfo.from((CompositeData) n.getUserData());
+                // Shenandoah GC also reports GC pauses, skip them
+                if (info.getGcName().startsWith("Shenandoah")) {
+                    if ("end of GC cycle".equals(info.getGcAction())) {
+                        gcCount++;
+                    }
```

Comment:
I think you want to do the same thing for ZGC here, which also reports both "pauses" and "cycles".

Reply:
Fixed. Thanks